lib/repack.c

```diff
@@ -1289,3 +1294,52 @@ repack_index_swap(PG_FUNCTION_ARGS)
 	SPI_finish();
 	PG_RETURN_VOID();
 }
+
+/**
+ * @fn      Datum get_table_and_inheritors(PG_FUNCTION_ARGS)
+ * @brief   Return array containing Oids of parent table and its children.
+ *
+ * get_table_and_inheritors(table)
+ *
+ * @param	table	parent table.
+ * @retval	regclass[]
+ */
+Datum
+repack_get_table_and_inheritors(PG_FUNCTION_ARGS)
+{
+	Oid			parent = PG_GETARG_OID(0);
+	List	   *relations;
+	Datum	   *relations_array;
+	int			relations_array_size;
+	ArrayType  *result;
+	ListCell   *lc;
+	int			i;
+
+	LockRelationOid(parent, AccessShareLock);
+
+	/* Check that parent table exists */
+	if (!SearchSysCacheExists1(RELOID, ObjectIdGetDatum(parent)))
+		PG_RETURN_ARRAYTYPE_P(construct_empty_array(OIDOID));
+
+	/* Also check that children exist */
+	relations = find_all_inheritors(parent, AccessShareLock, NULL);
```

Comment:
@MasahikoSawada, why? They'll be released automatically after transaction finishes. We're not going to benefit from releasing the locks right after we took them.

Should I add a comment above the function's definition?